Bluetooth: Mesh: Add Option config unprov beacon interval

LingaoM opened the pull request:
1 Add Option to specifies the second interval when device
send Unprovisioned Beacon.

2 Move BT_MESH_LPN_SUB_ALL_NODES_ADDR to LPN subgroup.

Signed-off-by: Lingao Meng <mengabc1086@gmail.com>

LingaoM committed:
Bluetooth: Mesh: Add Option config unprov beacon interval

Add Option to specifies the second interval when device
send Unprovisioned Beacon.

Signed-off-by: Lingao Meng <mengabc1086@gmail.com>

trond-snekvik commented on subsys/bluetooth/mesh/Kconfig:
```diff
@@ -29,6 +29,18 @@ config BT_MESH_PB_ADV
+if BT_MESH_PROV
+
+config BT_MESH_UNPROV_BEACON_INVL
+	int "The second interval to send the unallocated beacon"
```
```suggestion
	int "The second interval to send the unprovisioned beacon"
```

trond-snekvik commented on subsys/bluetooth/mesh/Kconfig:
```diff
@@ -29,6 +29,18 @@ config BT_MESH_PB_ADV
+	default 5
+	range 1 10
+	help
+	  This option specifies the second interval at which the
```
```suggestion
	  This option specifies the interval (in seconds) at which the
```

trond-snekvik commented on subsys/bluetooth/mesh/Kconfig:
```diff
@@ -29,6 +29,18 @@ config BT_MESH_PB_ADV
 
+if BT_MESH_PROV
+
+config BT_MESH_UNPROV_BEACON_INVL
```
Interval is generally shortened to `INT`, or the full `INTERVAL` is used.

trond-snekvik reviewed (changes requested):
Just a couple of minor comments :)

joerchan commented:
@LingaoM I think it would be best to have "2 Move BT_MESH_LPN_SUB_ALL_NODES_ADDR to LPN subgroup." as it's own commit.

LingaoM commented:
> @LingaoM It would be best to have "2 Move BT_MESH_LPN_SUB_ALL_NODES_ADDR to LPN subgroup." as it's own commit.
> 
> Can you please split it?

Done

trond-snekvik reviewed (changes requested):
Please fix the compilation error before merging. When kconfig dependencies are missing, the integer config entries aren't defined at all, so you'll need to replace the `IS_ENABLED(CONFIG_BT_MESH_PROV)` with an `#if`

joerchan commented:
@LingaoM Please check buildkite, there are compilation issues.

LingaoM commented:
> @LingaoM Please check buildkite, there are compilation issues.

Done

jhedberg commented on subsys/bluetooth/mesh/beacon.c:
```diff
@@ -285,10 +288,13 @@ static void beacon_send(struct k_work *work)
+
+		return;
 	}
+#if defined(CONFIG_BT_MESH_PROV)
```
Why is the dependency changed from BT_MESH_PB_ADV to BT_MESH_PROV?

jhedberg reviewed (changes requested):
See my inline comments.

Also, your first commit message still contains stuff about BT_MESH_LPN_SUB_ALL_ADDRS even though you moved that to the second commit.

LingaoM committed:
Bluetooth: Mesh：Move LPN_SUB_ALL to lpn subgroup

Move BT_MESH_LPN_SUB_ALL_NODES_ADDR to lpn subgroup

Signed-off-by: Lingao Meng <mengabc1086@gmail.com>

Commit:
Bluetooth: Mesh: Remove unnecessary #ifdefs from beacon code

Many #ifdefs can be removed, but the UNPROV_BEACON_INT Kconfig
variable needs to be also made available also. This is done by making
its prompt (user selectability) optional rather than the option
itself. This approach is fine for "parameter style" options, but
should probably not be used for feature enabling options.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

jhedberg commented:
@LingaoM I pushed a third patch on top of your two patches, which eliminates a bunch of unnecessary #ifdefs from the code (in favour of using IS_ENABLED). Could you take a look and confirm that it's fine with you?

LingaoM commented:
> @LingaoM I pushed a third patch on top of your two patches, which eliminates a bunch of unnecessary #ifdefs from the code (in favour of using IS_ENABLED). Could you take a look and confirm that it's fine with you?

LGTM
